### PR TITLE
fix: cleans up linter usage

### LINKED ISSUE #382: Review linters
##### What can be improved

As number of `//nolint[:]` comments is growing in the code it's time to review and fix or disable those which are "abused".

##### Why

Otherwise code gets clumsy.

### PR DESCRIPTION
**Pull request by bartoszmajsak**
#### Short description of what this resolves:

As we learnt in #378 our usage of golangi-ci linter was a bit wrong. This PR brings order and sanity :)

#### Changes proposed in this pull request:

- fix: disables hugeParam linter
- fix: proper usage of nolint comment
- fix: bypass gocyclo for few functions
- removes unused variable
- adds reasons behind every `//nolint` directive

For example `//nolint[:gosec]` is a wrong syntax (most likely from previous linter we had in place). With this syntax all linter checks are bypassed. The proper syntax is `//nolint:linter1,linter2 //reason explanation`.

Fixes #382

### CODE REVIEW

File: pkg/istio/destinationrule.go [hunk 1]
```diff
@@ -84,38 +76,38 @@ func DestinationRuleRevertor(ctx model.SessionContext, ref *model.Ref) error { /
...
 
-func mutateDestinationRule(dr istionetwork.DestinationRule, name string) (istionetwork.DestinationRule, error) { //nolint[:hugeParam]
+func mutateDestinationRule(dr istionetwork.DestinationRule, name string) istionetwork.DestinationRule {
 	dr.Spec.Subsets = append(dr.Spec.Subsets, &v1alpha3.Subset{
 		Name: name,
 		Labels: map[string]string{
 			"version": name,
 		},
 	})
-	return dr, nil
```
Comment (bartoszmajsak) [inline pkg/istio/destinationrule.go]: @aslakknutsen I removed error in few places as it was always `nil`

File: e2e/http_funcs.go [hunk 2]
```diff
@@ -14,7 +14,7 @@ func GetBody(rawURL string, cookies ...*http.Cookie) (string, error) {
 	for _, c := range cookies {
 		req.AddCookie(c)
 	}
-	resp, err := http.DefaultClient.Do(req) //nolint[:gosec]
+	resp, err := http.DefaultClient.Do(req) //nolint:gosec //reason test code
```
Comment (bartoszmajsak) [inline e2e/http_funcs.go]: `//nolint[:gosec]` is a wrong syntax (most likely from previous linter we had in place). With this syntax **all** linter checks are bypassed. The proper syntax is `//nolint:linter1,linter2 //reason explanation`. This PR fixes it for all existing ones.

File: test/cmd/test-service/main.go [hunk 3]
```diff
@@ -25,10 +25,6 @@ const (
 	EnvServiceCall = "SERVICE_CALL"
 )
 
-var (
-	rootDir = "test/cmd/test-service/assets/" //nolint[:deadcode]
```
Comment (aslakknutsen) [inline test/cmd/test-service/main.go]: This is required to use the dev mode for assets (reading from fs)

Comment (bartoszmajsak) [reply test/cmd/test-service/main.go]: Ah ok, the comment would help here :)

Comment (bartoszmajsak) [reply test/cmd/test-service/main.go]: Fixed in beed499

**Pull request merged**